flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactory.java

```diff
@@ -321,7 +321,9 @@ private static void validatePKConstraints(
                 && format.getChangelogMode().containsOnly(RowKind.INSERT)) {
             Configuration configuration = Configuration.fromMap(options);
             String formatName =
-                    configuration.getOptional(FactoryUtil.FORMAT).orElse(options.get(VALUE_FORMAT));
+                    configuration
+                            .getOptional(FactoryUtil.FORMAT)
+                            .orElse(configuration.get(VALUE_FORMAT));
```

Comment:
Does any of the existing tests cover the else branch?

Reply:
I tried adding another check by manipulating the options map in testPrimaryKeyValidation but it would always fail at some other validation already before reaching this code path

Reply:
Can't we add a test in `KafkaDynamicTableFactoryTest` to verify the exception includes the correct format? 

It should work ;)

Reply:
We have a test that uses the `FactoryUtil.FORMAT`, I tried adding the same test with passing in the `KafkaConnectorOptions.VALUE_FORMAT` but other validations always fail before we would reach this exception. I believe the code path is currently not reachable

Reply:
We should definitely investigate why this is not possible and check whether the value format configuration is actually used by someone. I could be that we broke it in 1.14 and no one noticed so far.

Reply:
I checked again and got it to work, maybe I forgot to set something else when I tried before